app/parse.py

```diff
@@ -9,4 +14,26 @@
 
 
 def get_all_courses() -> list[Course]:
-    pass
+    results = []
+    request_courses = requests.get(BASE_URL)
+    soup = BeautifulSoup(request_courses.text, "html.parser")
+    courses = soup.select(".ProfessionCard_cardWrapper__JQBNJ")
+    for course in courses:
+        results.append(
+            Course(
+                name=course.select_one("a.typography_landingH3__vTjok").text,
```

Comment:
Ensure that the CSS selector 'a.typography_landingH3__vTjok' matches the actual HTML structure of the webpage. If the structure changes, this could lead to an AttributeError if 'select_one' returns None.